File: Tarea5/Python/IMC.py
# Definición de las categorías
bajoPeso = "Bajo peso"
pesoNormal = "Peso normal"
sobrePeso = "Sobrepeso"

while True:
    # Mostrar las opciones disponibles al usuario
    print("---------Opciones disponibles:---------")
    print("1. Calcular IMC y mostrar resultados")
    print("2. Leer información de un registro")
    print("3. Borrar información de un registro")
    print("4. Salir del programa")
    print("---------------------------------------")
    
    # Leer la opción del usuario
    opcion = int(input("Ingrese la opción deseada: "))
    
    # Validar la opción
    if not (1 <= opcion <= 4):
        print("Opción no válida. Intente de nuevo.")
        continue
    
    if opcion == 1:
        nombre = input("Ingrese su nombre: ")
        peso = float(input("Ingrese su peso en kilogramos: "))
        altura = float(input("Ingrese su altura en metros: "))
        
        # Validar peso y altura
        if peso == 0 or altura == 0:
            print("Peso y altura no válidos, no es posible procesar el IMC.")
            continue
        
        # Calcular IMC
        imc = peso / (altura ** 2)
        
        # Determinar la categoría
        if imc < 18.5:
            categoria = bajoPeso
        elif imc < 24.9:
            categoria = pesoNormal
        else:
            categoria = sobrePeso
        
        # Mostrar resultados
        print(f"\nNombre: {nombre}")
        print(f"IMC: {imc:.2f}")
        print(f"Categoría: {categoria}")
        
        # Guardar en archivo
        with open('imc.txt', 'a') as f:
            f.write(f"Nombre: {nombre}, IMC: {imc:.2f}, Categoría: {categoria}\n")
        
        print("Información guardada en 'imc.txt'.")
    
    elif opcion == 2:
        try:
            with open('imc.txt', 'r') as f:
                contenido = f.readlines()
            
            if not contenido:
                print("El archivo está vacío, ingrese un registro.")
                continue
            
            nombre_buscado = input("Ingrese el nombre del registro que desea leer: ")
            encontrado = False
            
            for linea in contenido:
                if nombre_buscado in linea:
                    print(f"\nRegistro encontrado:\n{linea}")
                    encontrado = True
                    break
            
            if not encontrado:
                print(f"No se encontró ningún registro con el nombre '{nombre_buscado}'.")
        
        except FileNotFoundError:
            print("No se encontró el archivo 'imc.txt'.")
    
    elif opcion == 3:
        try:
            with open('imc.txt', 'r') as f:
                contenido = f.readlines()
            
            if not contenido:
                print("El archivo está vacío, ingrese un registro.")
                continue
            
            nombre_buscado = input("Ingrese el nombre del registro que desea borrar: ")
            encontrado = False
            lineas_nuevas = []
            
            for linea in contenido:
                if nombre_buscado not in linea:
                    lineas_nuevas.append(linea)
                else:
                    print(f"Registro encontrado y eliminado:\n{linea}")
                    encontrado = True
            
            with open('imc.txt', 'w') as f:
                f.writelines(lineas_nuevas)
            
            if not encontrado:
                print(f"No se encontró ningún registro con el nombre '{nombre_buscado}'.")
        
        except FileNotFoundError:
            print("No se encontró el archivo 'imc.txt'.")
    
    elif opcion == 4:
        print("¡Gracias por usar el programa!")
        break

File: Tarea5/Octave/IMC.m
% Definición de las categorías
bajoPeso = "Bajo peso";
pesoNormal = "Peso normal";
sobrePeso = "Sobrepeso";

while true
    % Mostrar las opciones disponibles al usuario
    disp("---------Opciones disponibles:---------");
    disp("1. Calcular IMC y mostrar resultados");
    disp("2. Leer información de un registro");
    disp("3. Borrar información de un registro");
    disp("4. Salir del programa");
    disp("---------------------------------------");
    % Leer la opción del usuario
    opcion = input("Ingrese la opción deseada: ");

    % Validar la opción
    if ~(1 <= opcion && opcion <= 4)
        disp("Opción no válida. Intente de nuevo.");
        continue;
    end

    if opcion == 1
        nombre = input("Ingrese su nombre: ", "s");
        peso = input("Ingrese su peso en kilogramos: ");
        altura = input("Ingrese su altura en metros: ");

        % Validar peso y altura
        if peso == 0 || altura == 0
            disp("Peso y altura no válidos, no es posible procesar el IMC.");
            continue;
        end

        % Calcular IMC
        imc = peso / (altura ^ 2);

        % Determinar la categoría
        if imc < 18.5
            categoria = bajoPeso;
        elseif imc < 24.9
            categoria = pesoNormal;
        else
            categoria = sobrePeso;
        end

        % Mostrar resultados
        fprintf("\nNombre: %s\n", nombre);
        fprintf("IMC: %.2f\n", imc);
        fprintf("Categoría: %s\n", categoria);

        % Guardar en archivo
        fid = fopen('imc.txt', 'a');
        fprintf(fid, "Nombre: %s, IMC: %.2f, Categoría: %s\n", nombre, imc, categoria);
        fclose(fid);

        disp("Información guardada en 'imc.txt'.");

    elseif opcion == 2
        if exist('imc.txt', 'file') == 2
            fid = fopen('imc.txt', 'r');
            contenido = textscan(fid, '%s', 'Delimiter', '\n');
            fclose(fid);

            if isempty(contenido{1})
                disp("El archivo está vacío, ingrese un registro.");
                continue;
            end

            nombre_buscado = input("Ingrese el nombre del registro que desea leer: ", "s");
            encontrado = false;

            for i = 1:length(contenido{1})
                if ~isempty(strfind(contenido{1}{i}, nombre_buscado))
                    fprintf("\nRegistro encontrado:\n%s\n", contenido{1}{i});
                    encontrado = true;
                    break;
                end
            end

            if ~encontrado
                fprintf("No se encontró ningún registro con el nombre '%s'.\n", nombre_buscado);
            end
        else
            disp("No se encontró el archivo 'imc.txt'.");
        end

    elseif opcion == 3
        if exist('imc.txt', 'file') == 2
            fid = fopen('imc.txt', 'r');
            contenido = textscan(fid, '%s', 'Delimiter', '\n');
            fclose(fid);

            if isempty(contenido{1})
                disp("El archivo está vacío, ingrese un registro.");
                continue;
            end

            nombre_buscado = input("Ingrese el nombre del registro que desea borrar: ", "s");
            encontrado = false;
            lineas_nuevas = {};

            for i = 1:length(contenido{1})
                if isempty(strfind(contenido{1}{i}, nombre_buscado))
                    lineas_nuevas{end+1} = contenido{1}{i};
                else
                    fprintf("Registro encontrado y eliminado:\n%s\n", contenido{1}{i});
                    encontrado = true;
                end
            end

            fid = fopen('imc.txt', 'w');
            for i = 1:length(lineas_nuevas)
                fprintf(fid, "%s\n", lineas_nuevas{i});
            end
            fclose(fid);

            if ~encontrado
                fprintf("No se encontró ningún registro con el nombre '%s'.\n", nombre_buscado);
            end
        else
            disp("No se encontró el archivo 'imc.txt'.");
        end

    elseif opcion == 4
        disp("¡Gracias por usar el programa!");
        break;
    end
end
